delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/stream/Operation.scala

```diff
@@ -82,23 +83,48 @@ sealed trait Operation { self =>
     override def outType: Typeable[Out] = self.inType
 
     override protected[stream] def asFs2: Pipe[Task, Elem[Operation.this.In], Elem[this.Out]] =
-      _.observe(self.asFs2.andThen(_.void))
+      _.chunks.evalTap { chunk =>
+        Stream.chunk(chunk).through(self.asFs2).compile.drain
+      }.flatMap(Stream.chunk)
+  }
+
+  /**
+    * Logs the elements of this stream as they are pulled.
+    *
+    * Logging is not done in `F` because this operation is intended for debugging,
+    * including pure streams.
+    */
+  def debug(formatter: Elem[Out] => String = (elem: Elem[Out]) => elem.toString,
```

Comment:
This is meant only for debugging purposes and it proved itself useful to do a:
```
operation.debug
```
To display the processed elems at the exit